### PR TITLE
updates on creating and updating lease

### PR DESCRIPTION
**Pull request by zhenz**
the following changes are made to updating lease:
- use horizon workflow
- if no physical host reservation, hide host min/max updates
the following changes are made to creating lease:
- reorganize static folder
- add checkbox for whether reserving the resource type
- disable inputs if resource type is not selected,
  including extra capability dropdowns
- use prev/next wizard to navigate among tabs, and check
  validity for each step before moving to the next
- prevent creating empty lease (no reservation)

### CODE REVIEW

File: blazar_dashboard/content/leases/templates/leases/_widget_computehost_capabilities.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div id="criteria-computehost" style="margin-bottom:50px">
+<div id="criteria-computehost" style="margin-bottom:50px" switchable_class="{{ widget.switchable_class }}">
```
Comment (diurnalist) [inline blazar_dashboard/content/leases/templates/leases/_widget_computehost_capabilities.html]: When you're doing something like this, data attributes are the way to go. I don't think it _really_ matters that much, but this is atypical HTML, and in any event, if you need to attach things to the DOM that are not related to presentation, data attributes are designed for this.

```html
<div data-switchable-class="{{ widget.switchable_class }}">
  ...
</div>
```

You can access data attributes with the `.dataset` property, and you access the attributes by referencing them by name. The browser exposes them in camelCase as opposed to hyphen-case:

```js
var switchableClass = div.dataset.switchableClass
```

File: blazar_dashboard/content/leases/workflows.py [hunk 2]
```diff
@@ -186,85 +193,144 @@ def prepare_datetimes(self, date_val, time_val):
...
         return conf.host_reservation.get("enabled", False)
 
 
 class SetNetworksAction(workflows.Action):
-    if conf.network_reservation.get('enabled'):
+    if conf.network_reservation.get('enabled', False):
+        with_network = forms.BooleanField(label=_("Reserve Network"),
+                                          initial=False,
+                                          required=False,
+                                          help_text=_("Please check if reserve "
```
Comment (diurnalist) [inline blazar_dashboard/content/leases/workflows.py]: I don't know that we need this help text, I don't think it adds much info.

File: blazar_dashboard/content/leases/workflows.py [hunk 3]
```diff
@@ -357,16 +491,25 @@ class CreateLease(workflows.Workflow):
                         'has been submitted.')
     failure_message = _('Unable to create the lease named "%s".')
     success_url = reverse_lazy('horizon:project:leases:index')
-    multipart = True
+    wizard = True
```
Comment (diurnalist) [inline blazar_dashboard/content/leases/workflows.py:486]: What's the difference between multipart and wizard?

**Review by diurnalist — APPROVED** [review]
A bit hard to review this w/o seeing it, it looks like this puts checkboxes on the first part of the workflow and those checkboxes control which future parts of the workflow the user sees?

**Comment by zhenz** [comment]
> A bit hard to review this w/o seeing it, it looks like this puts checkboxes on the first part of the workflow and those checkboxes control which future parts of the workflow the user sees?

I attached a snapshot of my devstack. It disables the inputs when unchecked. see below
Screen Shot 2021-07-13 at 11 22 44 AM

File: blazar_dashboard/content/leases/workflows.py (hunk 3 above)
Comment (zhenz) [reply blazar_dashboard/content/leases/workflows.py:486]: multipart is not necessary here. It's used for "multipart/form_data" for POST requests when uploading files. 
wizard is the “prev” and “next” buttons and validation after every step

**Pull request merged**